### PR TITLE
BC-3253 - Fix nuxtclient-ldap-deployment

### PR DESCRIPTION
**Pull request by BlexToGo**
# Short Description

Fix so that the `nuxtclient-ldap-deployment` on staging comes back up after #2345 

## Links to Ticket and related Pull-Requests

BC-3253

## Changes

Apply all the changes made to `ansible\roles\nuxt-client-core\templates\deployment.yml.j2` during #2345 also to `ansible\roles\nuxt-client-ldap\templates\deployment.yml.j2`

## Deployment

Only applicable on staging as other environments don't have `-ldap-deployment` deployments.

### CODE REVIEW

**Review by virgilchiriac — DISMISSED** [review]

File: ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2 [hunk 1]
```diff
@@ -38,13 +38,13 @@ spec:
...
         volumeMounts:
-        - mountPath: /app/http-headers.config.js
-          subPath: http-headers.js
-          name: http-headers
+        - mountPath: /var/run
+          name: pid-dir
+        - mountPath: /var/cache/nginx
+          name: cache-dir
+        - mountPath: /etc/nginx/conf.d
+          name: conf-dir
```
Comment (uidp) [inline ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2:47]: 👍

File: ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2 [hunk 2]
```diff
@@ -69,14 +69,14 @@ spec:
...
-          name: nuxtclient-configmap-files
-          items:
-          - key: http-headers.js
-            path: http-headers.js
+      - name: pid-dir
+        emptyDir: {}
+      - name: cache-dir
+        emptyDir: {}
+      - name: conf-dir
+        emptyDir: {}
```
Comment (uidp) [inline ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2:82]: 👍

**Review by uidp — CHANGES_REQUESTED** [review]

File: ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2 [hunk 3]
```diff
@@ -38,13 +38,13 @@ spec:
         envFrom:
         - configMapRef:
             name: nuxtclient-configmap
-        env:
-        - name: LEGACY_CLIENT_URL
```
[the remark this answers is not in the document]
Comment (BlexToGo) [reply ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2:42]: I just added `LEGACY_CLIENT_URL` again 👍 Though `ansible\roles\nuxt-client-core\templates\deployment.yml.j2` does not contain `LEGACY_CLIENT_URL`, could this be an issue or is `LEGACY_CLIENT_URL` just necessary for `nuxtclient-ldap-deployment`?
https://github.com/hpi-schul-cloud/nuxt-client/blob/main/ansible/roles/nuxt-client-core/templates/deployment.yml.j2#L38-L41

**Review by virgilchiriac — DISMISSED** [review]

File: ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2 (hunk 3 above)
[the remark this answers is not in the document]
Comment (uidp) [reply ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2:42]: Unfortunately I'm not that familiar with ansible. We need the `LEGACY_CLIENT_URL` env variable when starting the NGINX process of the nuxt-client pod. If that's the case we are fine.

**Review by uidp — DISMISSED** [review]

**Comment by BlexToGo** [comment]
Removed `LEGACY_CLIENT_URL` again after discussing it with @uidp because `nuxtclient-ldap-deployment` also uses the `nuxtclient-configmap` which already contains the `LEGACY_CLIENT_URL` environment variable because it's defined in `dof_app_deploy`
https://github.com/hpi-schul-cloud/dof_app_deploy/blob/e5b54f331988c2211cd6144862d2c5da80ef2d7f/ansible/group_vars/all/config.yml#L31-L35

File: ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2 (hunk 3 above)
[the remark this answers is not in the document]
Comment (BlexToGo) [reply ansible/roles/nuxt-client-ldap/templates/deployment.yml.j2:42]: Discussed this with @uidp, see latest commit and comment

**Review by uidp — APPROVED** [review]

**Review by Loki-Afro — APPROVED** [review]

**Pull request merged**